src/argilla/client/feedback/dataset/remote/dataset.py

```diff
@@ -0,0 +1,219 @@
+        return super()._iter_records()
+
+    @allowed_for_roles(roles=[UserRole.owner, UserRole.admin])
+    def add(
+        self,
+        records: Union[FeedbackRecord, Dict[str, Any], List[Union[FeedbackRecord, Dict[str, Any]]]],
+        show_progress: bool = True,
+    ) -> None:
+        """Pushes a list of `FeedbackRecord`s to Argilla.
+
+        Args:
+            records: can be a single `FeedbackRecord`, a list of `FeedbackRecord`,
+                a single dictionary, or a list of dictionaries. If a dictionary is provided,
+                it will be converted to a `FeedbackRecord` internally.
+            show_progress: Whether to show a `tqdm` progress bar while pushing the records.
+
+        Raises:
+            PermissionError: if the user does not have either `owner` or `admin` role.
+            Exception: If the pushing of the records to Argilla fails.
+        """
+        records = self._dataset._parse_and_validate_records(records)
+        for i in trange(
+            0, len(records), PUSHING_BATCH_SIZE, desc="Pushing records to Argilla...", disable=not show_progress
+        ):
+            records_batch = []
+            for record in records[i : i + PUSHING_BATCH_SIZE]:
+                if record.suggestions:
+                    for suggestion in record.suggestions:
+                        suggestion.question_id = self.__question_name2id[suggestion.question_name]
+                records_batch.append(
+                    record.dict(exclude={"id": ..., "suggestions": {"__all__": {"question_name"}}}, exclude_none=True)
+                )
+            datasets_api_v1.add_records(
+                client=self._client,
+                id=self._dataset_id,
+                records=records_batch,
+            )
+
+    @allowed_for_roles(roles=[UserRole.owner, UserRole.admin])
+    def delete(self, records: List[RemoteFeedbackRecord]) -> None:
+        """Deletes a list of `RemoteFeedbackRecord`s from Argilla.
+
+        Args:
+            records: A list of `RemoteFeedbackRecord`s to delete from Argilla.
+
+        Raises:
+            PermissionError: if the user does not have either `owner` or `admin` role.
+            RuntimeError: If the deletion of the records from Argilla fails.
+        """
+        num_records = len(records)
+        for start in range(0, num_records, DELETE_DATASET_RECORDS_MAX_NUMBER):
+            end = min(start + DELETE_DATASET_RECORDS_MAX_NUMBER, num_records)
+            try:
+                datasets_api_v1.delete_records(
+                    client=self._client,
+                    id=self._dataset_id,
+                    record_ids=[record.id for record in records[start:end]],
+                )
+            except Exception as e:
+                raise RuntimeError("Failed to remove records from Argilla") from e
+
+
+class RemoteFeedbackDataset(RemoteFeedbackDatasetBase[RemoteFeedbackRecords]):
```

Comment:
It should be possible to keep the `RemoteFeedbackDataset` as it was and extend the records class and move the filtered logic to a `FiltererdRecordsFeedbackDataset` class? Is any anything other than the filters and restricted method that is specific to this `FilteredFeedbackDataset`?

Reply:
Could you elaborate a little bit more? I don't quite understand the question 😩

Reply:
Forge this comment.